### PR TITLE
Overall Fixes

### PR DESCRIPTION
**Pull request by Darking360**
**Translation: Overall Fixes**

**Related:** [Original issue: Spanish Translation Final Tasks!](https://github.com/reactjs/es.reactjs.org/issues/158)

Overall checking on going, reading all core pages and submitting some changes and fixes.

### CODE REVIEW

File: content/docs/add-react-to-a-website.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ return (
 );
 ```
 
-Estos dos fragmentos de código son equivalentes. Mientras **JSX es [completamente opcional](/docs/react-without-jsx.html)**, muchas personas lo encuentran útil para escribir código relacionado con la IU -- ya sea usando React o con otras bibliotecas.
+Estos dos fragmentos de código son equivalentes. Mientras **JSX es [completamente opcional](/docs/react-without-jsx.html)**, muchas personas lo encuentran útil para escribir código relacionado con la interfaz de ssuario -- ya sea usando React o con otras bibliotecas.
```
Comment (tesseralis) [inline content/docs/add-react-to-a-website.md]: `usario`, not `ssuario`

File: content/docs/getting-started.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ Si vienes con experiencia como diseñador, [estos recursos](http://reactfordesig
 
 La documentación de React asume cierta familiaridad con la programación en Javascript. No necesitas ser un experto, pero es más difícil aprender React y Javascript al mismo tiempo.
 
-Recomendamos pasar por este [resumen de Javascript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript) para medir tu nivel de conocimiento. Te tomará entre 30 minutos a 1 hora, pero te sentirás más seguro aprendiendo React.
+Recomendamos pasar por este [resumen de Javascript](https://developer.mozilla.org/es-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript) para medir tu nivel de conocimiento. Te tomará entre 30 minutos a 1 hora, pero te sentirás más seguro aprendiendo React.
```
Comment (tesseralis) [inline content/docs/getting-started.md]: Is it `es-US` or just `es`?

File: content/docs/add-react-to-a-website.md (hunk 1 above)
Comment (Darking360) [reply content/docs/add-react-to-a-website.md]: My bad, thanks! 👏

File: content/docs/getting-started.md (hunk 2 above)
Comment (Darking360) [reply content/docs/getting-started.md]: In the code is `es-US` but when hitting the page, it redirects to `es`, I'll change it, and look for more. Thanks 🚀

**Comment by Darking360** [comment]
Hey guys 💪 I finished it 🎊 

So basically, I checked core pages, and some of the advanced ones The things that I encountered were these:

- Untranslated parts, or translated parts with the english version next to them.
- Code comments untranslated.
- Code that was translated, so I reverted those back to original.
- Typos and accents.
- Formatting issues, for example when they tried to add a bold font with ** they left a space after, so the format wasn't correct.
- Links broken in same pages, for example clicking a link to #callback-refs but that takes the user nowhere or top of the page.
- Links to docs in english, most of those changed to Spanish versions in wikipedia or MDN.

I think that was all, I'll certainly skipped one or two but that's most of it. Please take a look (I know it's a lot) but with this I think that we fix various issues 🚀 Thank you guys! I'll be looking for your feedback 📓 

**CC:** @alejandronanez @carburo @dmoralesm @tesseralis

**Comment by alejandronanez** [comment]
Holy cow, this is a huge (and amazing) PR - I’ll take a look at it tonight!

You’re a hero 🎖

**Comment by carburo** [comment]
I'll take a look later, probably tonight. Great job! 💪

File: examples/react-component-reference/get-snapshot-before-update.js [hunk 3]
```diff
@@ -5,8 +5,8 @@ class ScrollingList extends React.Component {
   }
 
   getSnapshotBeforeUpdate(prevProps, prevState) {
-    // Are we adding new items to the list?
-    // Capture the scroll position so we can adjust scroll later.
+    // Estamos agregando nuevos elementos a la lista?
```
Comment (carburo) [inline examples/react-component-reference/get-snapshot-before-update.js]:
```suggestion
    // ¿Estamos agregando nuevos elementos a la lista?
```

File: examples/react-component-reference/get-snapshot-before-update.js [hunk 4]
```diff
@@ -5,8 +5,8 @@ class ScrollingList extends React.Component {
...
+    // Captura la posición del scroll para que podamos ajustar el scroll despuës.
```
Comment (carburo) [inline examples/react-component-reference/get-snapshot-before-update.js]:
```suggestion
    // Captura la posición del scroll para que podamos ajustar el scroll después.
```

File: content/docs/reference-react-component.md [hunk 5]
```diff
@@ -169,7 +169,7 @@ Evita introducir cualquier efecto secundario o suscripciones en el constructor.
 
 > Nota
 >
-> **Evita copiar los props en el state! Es un error muy común:**
+> **Evita copiar los props en el estado! Es un error muy común:**
```
Comment (carburo) [inline content/docs/reference-react-component.md]:
```suggestion
> **¡Evita copiar los props en el estado! Es un error muy común:**
```

File: content/docs/reference-react-component.md [hunk 6]
```diff
@@ -179,11 +179,11 @@ Evita introducir cualquier efecto secundario o suscripciones en el constructor.
 >}
 >```
 >
-> El problema es que es innecesario (puedes usar `this.props.color` directamente en su lugar), esto crea errores (actualizaciones al prop `color` no se reflejarán en el state).
+> El problema es que es innecesario (puedes usar `this.props.color` directamente en su lugar), esto crea errores (actualizaciones al prop `color` no se reflejarán en el estado).
 >
-> **Sólo utiliza este patrón si deseas ignorar intencionalmente las actualizaciones de prop.** En ese caso, tiene sentido renombrar el prop a `initialColor` o `defaultColor`. Puedes forzar al componente a "limpiar" su state interno [ cambiando su `key`](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) cuando sea necesario.
+> **Sólo utiliza este patrón si deseas ignorar intencionalmente las actualizaciones de prop.** En ese caso, tiene sentido renombrar el prop a `initialColor` o `defaultColor`. Puedes forzar al componente a "limpiar" su estado interno [ cambiando su `key`](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) cuando sea necesario.
```
Comment (carburo) [inline content/docs/reference-react-component.md]:
```suggestion
> **Sólo utiliza este patrón si deseas ignorar intencionalmente las actualizaciones de prop.** En ese caso, tiene sentido renombrar el prop a `initialColor` o `defaultColor`. Puedes forzar al componente a "limpiar" su estado interno [cambiando su `key`](/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) cuando sea necesario.
```

File: content/docs/reference-react-component.md [hunk 7]
```diff
@@ -558,9 +558,9 @@ this.setState((state) => {
 
 Para más detalles, visite:
 
-* [State and Lifecycle guide](/docs/state-and-lifecycle.html)
-* [In depth: When and why are `setState()` calls batched?](https://stackoverflow.com/a/48610973/458193)
-* [In depth: Why isn't `this.state` updated immediately?](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
+* [Guía: Estado y ciclo de vida](/docs/state-and-lifecycle.html)
+* [En profundidad: Cuando y por qué las llamadas de `setState()` se hacen en lotes?](https://stackoverflow.com/a/48610973/458193)
+* [En profundidad: Por qué `this.state` no se actualiza inmediatamente?](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
```
Comment (carburo) [inline content/docs/reference-react-component.md]:
```suggestion
* [En profundidad: ¿Por qué `this.state` no se actualiza inmediatamente?](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
```

File: examples/forwarding-refs/fancy-button-simple-ref.js [hunk 8]
```diff
@@ -5,6 +5,6 @@ const FancyButton = React.forwardRef((props, ref) => (
   </button>
 ));
 
-// You can now get a ref directly to the DOM button:
+// Ahora puedes obtener una referencia directa al boton del DOM:
```
Comment (carburo) [inline examples/forwarding-refs/fancy-button-simple-ref.js]:
```suggestion
// Ahora puedes obtener una referencia directa al botón del DOM:
```

File: content/docs/state-and-lifecycle.md [hunk 9]
```diff
@@ -31,7 +31,7 @@ setInterval(tick, 1000);
 
 [**Pruébalo en CodePen**](http://codepen.io/gaearon/pen/gwoJZk?editors=0010)
 
-En esta sección, aprenderemos como hacer al componente `Clock` verdaderamente reutilizable y encapsulado. Configurará su propio temporizador y se actualizará cada segundo.
+En esta sección, aprenderemos como hacer al componente `Clock` verdaderamente reutilizable y encapsulado. Configurarás tu propio temporizador y se actualizará cada segundo.
```
[the remark this answers is not in the document]
Comment (Darking360) [reply content/docs/state-and-lifecycle.md]: Mala mía acá por no referenciar la traducción original, tienes toda la razón 💪

File: content/docs/code-splitting.md [hunk 10]
```diff
@@ -202,7 +202,7 @@ const MyComponent = () => (
 
 Decidir en qué parte de tu aplicación introducir la división de código puede ser un poco complicado. Quieres asegurarte de elegir lugares que dividan los *bundles* de manera uniforme, sin interrumpir la experiencia del usuario.
 
-Un buen lugar para comenzar es con las rutas. La mayoría de la gente en la web está acostumbrada a que las transiciones entre páginas se tomen cierto tiempo en cargar. También tiendes a rerenderizar todo de una vez, así que es improbable que tus usuarios interactúen con otros elementos en la página al mismo tiempo.
+Un buen lugar para comenzar es con las rutas. La mayoría de la gente en la web está acostumbrada a que las transiciones entre páginas se tomen cierto tiempo en cargar. También tiendes a re-renderizar todo de una vez, así que es improbable que tus usuarios interactúen con otros elementos en la página al mismo tiempo.
```
[the remark this answers is not in the document]
Comment (Darking360) [reply content/docs/code-splitting.md]: Perfecto, devolveré al original y revisare cuales otros revertí 🤔

**Comment by Darking360** [comment]
Greetings guys! I just added @carburo feedback.

If you could take a last look before merging this would be awesome!

**CC:** @alejandronanez @tesseralis @dmoralesm

File: content/docs/add-react-to-a-website.md [hunk 11]
```diff
@@ -147,7 +147,7 @@ return (
 );
 ```
 
-Estos dos fragmentos de código son equivalentes. Mientras **JSX es [completamente opcional](/docs/react-without-jsx.html)**, muchas personas lo encuentran útil para escribir código relacionado con la IU -- ya sea usando React o con otras bibliotecas.
+Estos dos fragmentos de código son equivalentes. Mientras **JSX es [completamente opcional](/docs/react-without-jsx.html)**, muchas personas lo encuentran útil para escribir código relacionado con la interfaz de usuario -- ya sea usando React o con otras bibliotecas.
```
Comment (alejandronanez) [inline content/docs/add-react-to-a-website.md]: Bravo!

File: content/docs/create-a-new-react-app.md [hunk 12]
```diff
@@ -33,7 +33,7 @@ El equipo de React principalmente recomienda las siguientes soluciones:
 - Si estás **aprendiendo React** o **creando una nueva [aplicación de página única](/docs/glossary.html#single-page-application),** usa [Create React App](#create-react-app).
 - Si estás construyendo un **sito web renderizado en servidor con Node.js,** prueba [Next.js](#nextjs).
 - Si estás construyendo un **sitio web orientado a contenido estático,** prueba [Gatsby](#gatsby).
-- Si estás construyendo una **biblioteca de componentes** o **integrando una base de código existente**, prueba [Cadenas de Herramientas más Flexibles](#cadenas-de-herramientas-ms-flexibles).
+- Si estás construyendo una **biblioteca de componentes** o **integrando una base de código existente**, prueba [Cadenas de Herramientas más Flexibles](#more-flexible-toolchains).
```
Comment (alejandronanez) [inline content/docs/create-a-new-react-app.md]: Good catch!

**Pull request merged**